src-built-in/components/toolbar/stores/toolbarStore.js

```diff
@@ -171,16 +171,19 @@ class _ToolbarStore {
 			done();
 		});
 		let onBoundsSet = (bounds) => {
-			debugger
 			bounds = bounds.data ? bounds.data : bounds;
 			self.Store.setValue({ field: "window-bounds", value: bounds });
 			FSBL.Clients.WindowClient.setComponentState({
 				field: 'window-bounds',
 				value: bounds
 			}, Function.prototype);
 		}
-
-		FSBL.Clients.WindowClient.finsembleWindow.addListener("bounds-change-end", onBoundsSet)
+		let restoreWindow = (e) => {
+			finsembleWindow.restore();
+		}
+		//Immediately restore on maximize.
+		finsembleWindow.addListener("maximized", restoreWindow);
```

Comment:
@carterbs I think this is what's happening? "to prevent double click from maximizing, we instantly restore the window whenever it is maximized".

If so, there ought to be a comment stating explicitly what is happening here and why (eliminate campfire knowledge).

We might want to turn this into a window capability (allowMaximize:false, allowMinimize:false).

Reply:
I think you could just commit directly to the branch.